crates/db_views/src/combined/report_combined_view.rs

```diff
@@ -324,17 +327,24 @@ impl ReportCombinedQuery {
       );
     }
 
-    if let Some(post_id) = self.post_id {
-      query = query.filter(post::id.eq(post_id));
+    if user.local_user.admin {
+      // By default admins only see reports with `violates_instance_rules == true`, for communities
+      // where they are moderator, or reports which have been unresolved for more than 3 days.
+      // With parameter `show_mod_reports` they can also see newer reports in other communities.
+      let show_mod_reports = self.show_mod_reports.unwrap_or_default();
+      if !show_mod_reports {
+        query = query.filter(filter_visible_to_admins(Utc::now() - Days::new(3)));
+      }
+    } else {
+      // Mods can only see reports for communities where they are moderator, and
+      // which have `violates_instance_rules == false`.
+      query = query
+        .filter(filter_is_mod())
+        .filter(filter_violates_instance_rules().is_distinct_from(true));
```

Comment:
I can barely follow this, because so many types are being filtered out (sometimes not, with `.or` filters) within these, and duplicated `filter_is_mod()` in multiple levels. 

This would be much easier to follow:

```
if !admin {
  filter_is_mod()
else {
  if !show_mod_reports {
    not(filter_show_mod_reports)
  }
}
```

Reply:
Cleaned this up and added more comments.